packages/loopring_v3/contracts/amm/libamm/AmmStatus.sol

```diff
@@ -71,16 +71,26 @@ library AmmStatus
             }));
         }
 
-        // Mint all liquidity tokens to the pool account on L2
-        S.balanceOf[address(this)] = AmmData.POOL_TOKEN_MINTED_SUPPLY;
-        S.allowance[address(this)][address(exchange.getDepositContract())] = type(uint256).max;
-        exchange.deposit(
-            address(this), // from
-            address(this), // to
-            address(this), // token
-            uint96(AmmData.POOL_TOKEN_MINTED_SUPPLY),
-            new bytes(0)
-        );
+        bool newPool = (S.accountID == 0);
+        if (newPool) {
+            S.sharedConfig = IAmmSharedConfig(config.sharedConfig);
+            S.exchange = exchange;
+            S.exchangeOwner = exchange.owner();
+            S.exchangeDomainSeparator = exchange.getDomainSeparator();
+            S.accountID = config.accountID;
+            S.poolTokenID = exchange.getTokenID(address(this));
+
+            // Mint all liquidity tokens to the pool account on L2
+            S.balanceOf[address(this)] = AmmData.POOL_TOKEN_MINTED_SUPPLY;
+            S.allowance[address(this)][address(exchange.getDepositContract())] = type(uint256).max;
+            exchange.deposit(
+                address(this), // from
+                address(this), // to
+                address(this), // token
+                uint96(AmmData.POOL_TOKEN_MINTED_SUPPLY),
+                new bytes(0)
+            );
+        }
```

Comment:
in method `_shutdown`, shall we assert `S.shutdownTimestamp == 0` ?

Reply:
`shutdown` and `shutdownByController` are already `onlyWhenOnline`.